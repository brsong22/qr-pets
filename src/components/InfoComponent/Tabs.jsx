import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Tab from './Tab';

class Tabs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      onTab: 'info',
    };
  }

  render() {
    const { tabs, onTab } = this.props;
    const t = tabs.map(tab => (
      <Tab tabText={tab} onTab={onTab} />
    ));
    return (
      <div className="tabs">
        {t}
      </div>
    );
  }
}

Tabs.propTypes = {
  tabs: PropTypes.arrayOf(PropTypes.string),
  onTab: PropTypes.string,
};

Tabs.defaultProps = {
  tabs: [],
  onTab: 'info',
};

export default Tabs;
