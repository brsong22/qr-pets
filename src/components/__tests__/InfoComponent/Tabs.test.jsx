
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tabs from '../../InfoComponent/Tabs';

describe('Tabs', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<Tabs />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
