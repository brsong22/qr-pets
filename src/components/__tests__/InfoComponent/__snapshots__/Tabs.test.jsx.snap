// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders as expected 1`] = `
<div
  className="tabs"
/>
`;
